test/test_utils/__init__.py

```diff
@@ -1510,6 +1510,35 @@ def get_python_version_from_image_uri(image_uri):
     python_version = python_version_search.group()
     return "py36" if python_version == "py3" else python_version
 
+def construct_buildspec_path(dlc_path, framework_path, buildspec, framework_version):
```

Comment:
any reason to add this

Reply:
Address an issue with deriving framework short version. Previous implementation introduced a bug with the framework_short_version having a NoneType for AG.